src/token_utilities.py

```diff
@@ -20,8 +20,8 @@ def get_cache(cache_key):
 
 def process_token(token):
     parts = token.split('.')
-    if len(parts) == 3:
-        try:
+    try:
+        if len(parts) == 3:
```

Comment:
This is a redundant if statement, I've said previously to handle this as a ValueError and catch it. You're already in the try...